src/test/java/com.teragrep.jos_01.procfs/ProcessTest.java

```diff
@@ -0,0 +1,355 @@
+            Assertions.assertNotNull(stat.delayacct_blkio_ticks());
+            Assertions.assertNotNull(stat.guest_time());
+            Assertions.assertNotNull(stat.cguest_time());
+            Assertions.assertNotNull(stat.start_data());
+            Assertions.assertNotNull(stat.end_data());
+            Assertions.assertNotNull(stat.start_brk());
+            Assertions.assertNotNull(stat.arg_start());
+            Assertions.assertNotNull(stat.arg_end());
+            Assertions.assertNotNull(stat.env_start());
+            Assertions.assertNotNull(stat.env_end());
+            Assertions.assertNotNull(stat.exit_code());
+
+        });
+    }
+
+    // Statm status object should contain all of the listed fields.
+    @Test
+    public void statmTest() {
+        Assertions.assertDoesNotThrow(() -> {
+            Process process = new Process(1);
+            Statm statm = process.statm();
+            Assertions.assertNotNull(statm.size());
+            Assertions.assertNotNull(statm.resident());
+            Assertions.assertNotNull(statm.shared());
+            Assertions.assertNotNull(statm.text());
+            Assertions.assertNotNull(statm.lib());
+            Assertions.assertNotNull(statm.data());
+            Assertions.assertNotNull(statm.dt());
+        });
+    }
+
+    // Timestamps should be available and should be different if the proc method is called later
+    @Test
+    public void timestampTest() {
+        Assertions.assertDoesNotThrow(() -> {
+            systemd = new Process(1);
+            Stat stat = systemd.stat();
+            Stat stat2 = systemd.stat();
+            Statm statm = systemd.statm();
+
+            // Timestamps should always have a value
+            Assertions.assertTrue(stat.timestamp() != null);
+            Assertions.assertTrue(statm.timestamp() != null);
+
+            // Timestamps should be different if called at different times, even when calling the same method again
+            Assertions.assertFalse(stat.timestamp().equals(stat2.timestamp()));
+            Assertions.assertFalse(stat.timestamp().equals(statm.timestamp()));
+
+        });
+    }
+
+    // Get the JVM process this test is running in, and make two delayed calls to cpuTime to make sure cpuTime actually increments as process is in use
+    // Uses SysconfInterface.Fake() to get a hardcoded CPU tick rate (100) without actually having to compile Native C code.
+    @Test
+    public void CpuTimeTest() {
+        Assertions.assertDoesNotThrow(() -> {
+            long pid = Long.parseLong(ManagementFactory.getRuntimeMXBean().getName().split("@")[0]);
+            Process jvm = new Process(pid, new LinuxOS(new SysconfInterface.Fake()));
+            float cpuTime1 = jvm.cpuTime();
+            ArrayList<Double> workArray = new ArrayList<Double>();
+            for (int i = 0; i < 30000000; i++) { // Needs to do a lot of work to have different CPUtimes since the times are in seconds, not ticks
```

Comment:
Seems like this will take at least 240MB of memory, shouldn't be an issue though